webapp/src/js/services/telemetry.js

```diff
@@ -94,13 +94,42 @@ angular
     };
 
     var generateMetadataSection = function() {
-      var date = moment(getLastAggregatedDate());
-      return {
-        year: date.year(),
-        month: date.month(),
-        user: Session.userCtx().name,
-        deviceId: getUniqueDeviceId(),
-      };
+      return $q.all([
+          DB().get('_design/medic-client'),
+          DB().query('medic-client/forms', {include_docs: true})
+      ]).then(([ddoc, formResults]) => {
+        const date = moment(getLastAggregatedDate());
+        const version = (ddoc.deploy_info && ddoc.deploy_info.version) || 'unknown';
+        const forms = formResults.rows.reduce((keyToVersion, row) => {
+          let version;
+          if (row.doc._attachments && row.doc._attachments.xml) {
+            // While XML forms may have a version declared in the XML as of writing they are not
+            // required, seldom used, are not enforced (they don't have to change if the form changes)
+            // and so are entirely unreliable. The hash is the only real safe version we have.
```

Comment:
From your comment on the issue:

> For xml forms I've used the md5 digest hash: _rev doesn't work for us because PouchDB just makes them up...

Really? I think the rev on the Couch and Pouch should be the same otherwise how does conflict detection work? When I check locally the _rev is the same in Pouch as Couch - did I just get lucky?

If we can, I'd prefer to use revs because it's a little easier to check.

Reply:
Hmmmmmmmmmmmmmmmmmmmmmmmmmmm

We both get to be right here. Democracy!

You are right in the sense that if you replicate `_rev: "1-abc"` you will write it as `1-abc`. I am right in the sense that if the first write (where a `_rev` needs to generated) is on PouchDB the resulting rev hash is random (as opposed to CouchDB where it is a hash of the contents).

So yeah, we should use _rev because we get a nice number at the front, and at least for now all form writes happen directly into CouchDB so the hashes should make sense as well.